x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/dataframe/transforms/DataFrameTransformState.java

```diff
@@ -33,28 +33,44 @@
 public class DataFrameTransformState implements Task.Status, PersistentTaskState {
     public static final String NAME = DataFrameField.TASK_NAME;
 
-    private final IndexerState state;
+    private final DataFrameTransformTaskState state;
+    private final IndexerState indexerState;
     private final long generation;
 
     @Nullable
     private final SortedMap<String, Object> currentPosition;
+    @Nullable
+    private final String reason;
 
-    private static final ParseField STATE = new ParseField("transform_state");
+    private static final ParseField STATE = new ParseField("state");
```

Comment:
`DataFrameTransformStateAndStats` has this class as a member under `state` so the path will be `state.state` which looks a bit funny.

Reply:
yeah :( any better name is helpful. Also, keeping the `indexer_state` as `transform_state` is not really cool to me either. 

What about `task_state` ? Or we rename in `DataFrameTransformStateAndStats` so that it is `task_state.state`...

Reply:
@hendrikmuhs ^^^ What do you think?

Reply:
It looks odd to me to have 2 states at least from a high-level perspective, lets revisit what we need in stats. When implementing the 1st version I simply added everything possible at the time of writing, but I considered simplifying it. `indexer_state` is a low-level implementation detail. The afterKey is even more internal detail without being useful for a user (in rollups its the last time bucket rollup has processed, but for data frames it is some pagination key).

If I get it right this PR introduces a start/stopped/failed high-level state. So we have everything we need, except one thing: We do not know whether the data frame is currently "active", meaning a new checkpoint is being generated or with other words: the indexer is `indexing`. If we could add this, maybe we can get rid of indexer_state. I do not suggest adding a new state to wrap `indexing`, but it could be a new field: `task: active/waiting`. Hopefully soon we also add some progress for the current run.

However: `indexer_state` might be helpful when debugging a problem, so maybe we can hide it default but show it on-demand. The API's for retrieving index stats do something similar with a `level` query parameter (for checkpoints I am thinking in the same direction as I do not want to expose all detail per default).

Thoughts?

Reply:
another option: as this is already the state object it could be `task` and `indexer` so that `DataFrameTransformStateAndStats` returns:

{
"state": {
  "task": "started",
  "indexer": "indexing"
  }
}